lib/active_merchant/billing/gateways/ixopay.rb

```diff
@@ -67,55 +54,193 @@ def scrub(transcript)
+            xml.tag! 'company', options[:billing_address][:company]
+          end
+
+          xml.tag! 'email', options[:email]
+          xml.tag! 'ipAddress', options[:ip] || '127.0.0.1'
+        end
       end
 
-      def add_invoice(post, money, options)
-        post[:amount] = amount(money)
-        post[:currency] = (options[:currency] || currency(money))
+      def add_billing_address(xml, address)
+        if address[:name]
+          xml.tag! 'firstName', split_names(address[:name])[0]
+          xml.tag! 'lastName',  split_names(address[:name])[1]
+        end
+
+        xml.tag! 'billingAddress1', address[:address1]
+        xml.tag! 'billingAddress2', address[:address2]
+        xml.tag! 'billingCity',     address[:city]
+        xml.tag! 'billingPostcode', address[:zip]
+        xml.tag! 'billingState',    address[:state]
+        xml.tag! 'billingCountry',  address[:country]
+        xml.tag! 'billingPhone',    address[:phone]
       end
 
-      def add_payment(post, payment)
+      def add_shipping_address(xml, address)
+        if address[:name]
+          xml.tag! 'shippingFirstName', split_names(address[:name])[0]
+          xml.tag! 'shippingLastName',  split_names(address[:name])[1]
+        end
+
+        xml.tag! 'shippingCompany',   address[:company]
+        xml.tag! 'shippingAddress1',  address[:address1]
+        xml.tag! 'shippingAddress2',  address[:address2]
+        xml.tag! 'shippingCity',      address[:city]
+        xml.tag! 'shippingPostcode',  address[:zip]
+        xml.tag! 'shippingState',     address[:state]
+        xml.tag! 'shippingCountry',   address[:country]
+        xml.tag! 'shippingPhone',     address[:phone]
       end
 
-      def parse(body)
-        {}
+      def new_transaction_id
+        SecureRandom.uuid
       end
 
-      def commit(action, parameters)
+      def commit(action, request, options={})
         url = (test? ? test_url : live_url)
-        response = parse(ssl_post(url, post_data(action, parameters)))
+
+        begin
+          raw_response = ssl_post(url, request, headers(request))
+        rescue StandardError => error
+          return response_from_request_error(action, error)
+        end
+
+        response = parse(action, raw_response)
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      def response_from_request_error(action, error)
+        response = parse(action, error.response.body)
 
         Response.new(
           success_from(response),
           message_from(response),
           response,
           authorization: authorization_from(response),
-          avs_result: AVSResult.new(code: response["some_avs_response_key"]),
-          cvv_result: CVVResult.new(response["some_cvv_response_key"]),
           test: test?,
```

Comment:
Why were the AVS and CVV deleted? Are they just not included in the response from the gateway?

Reply:
Correct -- they are not present in the gateway response.